client_test/rpc_test.go

```diff
@@ -419,21 +425,26 @@ func checkNotifications[T channelInfo](t *testing.T, client string, required []T
 		select {
 		case info := <-notifChan:
 
-			js := marshalToJson(t, info)
+			notifJSON := marshalToJson(t, info)
 			t.Logf("%s received %v+", client, info)
 
 			// Check that the notification is a required or optional one.
-			_, found := acceptableNotifications[js]
+			_, found := acceptableNotifications[notifJSON]
 			if !found {
-				t.Fatalf("%s received unexpected notification: %v", client, info)
+				unexpectedNotifications[notifJSON] = true
 			}
 			// To signal we received a notification we delete it from the map
-			delete(acceptableNotifications, js)
+			delete(acceptableNotifications, notifJSON)
```

Comment:
This line will run even if `found=false`. What happens if you try and delete a key that doesn't exist?

Regardless, it might be nicer to do 

```go
if found {
    delete(acceptableNotifications, notifJSON)
} else {
    unexpectedNotifications[notifJSON] = true
}


```

Reply:
That delete of a non-key is a noop, but your suggestion is better anyhow.

7fdd7be1